src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpSimplifyTypeNamesDiagnosticAnalyzer.cs

```diff
@@ -40,8 +72,10 @@ protected override void AnalyzeSemanticModel(SemanticModelAnalysisContext contex
             var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult()!;
             var root = syntaxTree.GetRoot(cancellationToken);
 
-            var simplifier = new TypeSyntaxSimplifierWalker(this, semanticModel, optionSet, cancellationToken);
+            var simplifier = new TypeSyntaxSimplifierWalker(this, semanticModel, optionSet, ignoredSpans: codeBlockIntervalTree, cancellationToken);
```

Comment:
consider making teh whole walker poolable.

Reply:
Memory savings would not exceed 0.012%.

Reply:
it's not about memory savings for me.  It's about simplifying the logic around the lazy `new list` that you have inside the walker.  it means the walker doesn't have to care about htat :)

Reply:
Right, but it has other downsides:

* Fields not marked `readonly`
* New code to handle initialization and clearing
* New code at use sites to free the pooled object

The diagnostics list change was easier to constrain.

Reply:
hrmm.. comment seems t ohave been lost.  it's not about the memory savings here.  it's about making teh code clearer below wrt to havin to lazily create the diagnostics list.

Reply:
The benefits of not having to lazily initialize the diagnostics list are negated (and then some) by multiple complexities introduced by making this type capable of pooling. Considering how minimal the lazy initialization overhead is in this type, I don't see a compelling reason to make this change.